src/app/core/services/cart.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+
+import { DataService } from './data.service';
+import { IProduct, ICartProduct } from 'src/app/shared/interfaces/product';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class CartService {
+  private cartIds: string[] = [];
```

Comment:
as I understand, currently it's not ids only. I should be ICartProduct[]. Please rename to cartProducts, and refactor all references